src/Nethermind/Nethermind.Consensus/Producers/PayloadAttributes.cs

```diff
@@ -54,7 +60,10 @@ public static class PayloadAttributesExtensions
     public static string ComputePayloadId(this PayloadAttributes payloadAttributes, BlockHeader parentHeader)
     {
         bool hasWithdrawals = payloadAttributes.Withdrawals is not null;
-        Span<byte> inputSpan = stackalloc byte[32 + 32 + 32 + 20 + (hasWithdrawals ? 32 : 0)];
+        bool hasBeaconStateRoot = payloadAttributes.BeaconStateRoot is not null;
+        const int preambleLength = 32 + 32 + 32 + 20;
+
+        Span<byte> inputSpan = stackalloc byte[preambleLength + (hasWithdrawals ? 32 : 0) + (hasBeaconStateRoot ? 32 : 0)];
```

Comment:
if we have a `V3` subclass this becomes virtual call, check what we did in https://github.com/NethermindEth/nethermind/tree/optimism branch with it